app/jobs/users/self_delete_job.rb

```diff
@@ -0,0 +1,15 @@
+module Users
+  class SelfDeleteJob < ApplicationJob
+    queue_as :users_self_delete
+
+    def perform(user_id, service = Users::Delete)
+      user = User.find_by(id: user_id)
+      return unless user
+
+      service.call(user)
+      NotifyMailer.account_deleted_email(user).deliver
+    rescue StandardError => e
```

Comment:
I feel bad for rescuing `StandardError`, if you have better ideas, please share.
I don't have access to logs, so I'm not sure if anyone looks into them to notice problems :woman_shrugging: